### ISSUE TITLE
Running buildah rootless inside a container

### ISSUE THREAD

**Issue by Logibox** [containers/buildah#1480]
**Description**

This is more of a question than a bug report - is it possible to run buildah rootless inside of a container? From my initial experiments, I'm guessing not - but I'm unsure if it's a misconfiguration on my end, a limitation of buildah or a bug. Thanks!

**Steps to reproduce the issue:**
```
$ cat Dockerfile
FROM fedora:29

RUN dnf -y install buildah

RUN groupadd --gid 1000 somebody && \
    useradd --uid 1000 --gid 1000 somebody
$ buildah bud -t=fedora-buildah .
... <omitted> ...
$ podman run -u 1000:1000 -it localhost/fedora-buildah buildah 
Error: error running newgidmap: exit status 1: newgidmap: write to gid_map failed: Operation not permitted

ERRO[0000] exit status 1       
```

This seems to occur regardless of whether the 'outer' podman is running rootless or not.

**Output of `rpm -q buildah` or `apt list buildah`:**

```
$ rpm -q buildah
buildah-1.6-2.git5f95bd9.fc29.x86_64
$ podman run -it localhost/fedora-buildah rpm -q buildah
buildah-1.7-2.git873f001.fc29.x86_64
```

**Output of `buildah version`:**

```
(paste your output here)
```

**Output of `podman version` if reporting a `podman build` issue:**

```
(paste your output here)
```

**Output of `cat /etc/*release`:**

```
(paste your output here)
```

**Output of `uname -a`:**

```
(paste your output here)
```

**Output of `cat /etc/containers/storage.conf`:**

```
(paste your output here)
```

**Comment by rhatdan** [comment]
I think it should be possible to run buildah within a rootless container.  We are working on better ways of running buildah in a locked down container.

1 First step is to run it in a container without SYS_ADMIN Privs.
2 Next Step is to run it in a User Namespaced Container without SYS_ADMIN Privs.

In order to get what you are trying to do, to work you would need to run a --privileged container.

Bottom line is your goal to run a locked down build #2 is probably better then what you are trying.

**Comment by Logibox** [comment]
Thanks @rhatdan `--privileged` fixes the issue:
```
$ podman run --privileged -u 1000:1000 -it localhost/fedora-buildah buildah
A tool that facilitates building OCI images

Usage:
  buildah [flags]
  buildah [command]
...
```
Basically the issue is that we have a policy that none of the containers in our CI system run as UID 0 (as we're still using docker with no user-namespace support in our infrastructure), so buildah would need to run rootless in this scenario. I've tested this with docker and I'm satisfied with the result.

However I still get the error when podman itself is running rootless:
```
$ id
uid=1000(myates) gid=1000(myates) groups=1000(myates),10(wheel),1001(docker) context=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023
$ podman run --privileged -u 1000:1000 -it localhost/fedora-buildah buildah
Error: error running newgidmap: exit status 1: newgidmap: write to gid_map failed: Operation not permitted

ERRO[0000] exit status 1                                
```
Is that expected?

**Comment by rhatdan** [comment]
Yes because you are attempting to create a usernamespace while in a user namespace and that is blocked.

**Comment by Logibox** [comment]
Ah ok. Thanks for looking into that. My original issue is solved so I'll close this now.
